#Bioinformatic Analysis Software


###[PacBio Assembly Software Options] (https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Large-Genome-Assembly-with-PacBio-Long-Reads)

| Programs       | Description    | Source
| :------------- | :------------- | :------------- |
| SMRT Analysis  | Software suite is designed for use with Single Molecule, Real-Time (SMRT) Sequencing data.| [Link] (http://www.pacb.com/support/software-downloads)
| Celera Assembler| Celera Assembler is a de novo whole-genome shotgun (WGS) DNA sequence assembler, and can use any combination of platform reads. |
| | |
| | |
| MUMmer         | A system for rapidly aligning entire genomes, whether in complete or draft form. | [Link] (http://mummer.sourceforge.net/)
| Mauve          | A system for constructing multiple genome alignments in the presence of large-scale evolutionary events such as rearrangement and inversion.	| [Link] (http://darlinglab.org/mauve/mauve.html)
| Velvet | Short read de novo assembler using de Bruijn graphs | [Link] (http://www.ebi.ac.uk/~zerbino/velvet/)
| Cerulean | Cerulean extends contigs assembled using short read datasets like Illumina paired-end reads using long reads like PacBio RS long reads. | [Link] (https://sourceforge.net/projects/ceruleanassembler/) |
| PBJelly | PBJelly is a highly automated pipeline that aligns long sequencing reads (such as PacBio RS reads or long 454 reads in fasta format) to high-confidence draft assembles. PBJelly fills or reduces as many captured gaps as possible to produce upgraded draft genomes. | [Link] (https://sourceforge.net/projects/pb-jelly/)
| PBHoney | PBHoney is an implementation of two variant-identification approaches designed to exploit the high mappability of long reads (i.e., greater than 10,000 bp). PBHoney considers both intra-read discordance and soft-clipped tails of long reads to identify structural variants. | [Link] (https://sourceforge.net/projects/pb-jelly/)
| Sprai | Sprai (single-pass read accuracy improver) is a tool to correct sequencing errors in single-pass reads for de novo assembly. | [Link] (http://zombie.cb.k.u-tokyo.ac.jp/sprai/README.html)
